pkg/decoder/decoder.go

```diff
@@ -9,100 +9,128 @@ import (
 	"bytes"
 
 	"github.com/DataDog/datadog-log-agent/pkg/config"
-	"github.com/DataDog/datadog-log-agent/pkg/message"
 )
 
-// Payload represents a list of bytes and an optional reference to its origin
-type Payload struct {
+// contentLenLimit represents the length limit above which we want to truncate the output content
+var contentLenLimit = 256 * 1000
+
+// Input represents a list of bytes consumed by the Decoder
+type Input struct {
 	content []byte
-	offset  int64
 }
 
-// NewPayload returns a new decoder payload
-func NewPayload(content []byte, offset int64) *Payload {
-	return &Payload{content, offset}
+// NewInput returns a new input
+func NewInput(content []byte) *Input {
+	return &Input{content}
+}
+
+// Output represents a list of bytes produced by the Decoder
+type Output struct {
+	Content    []byte
+	ContentLen int
+	ShouldStop bool
 }
 
-// Decoder splits raw data based on `\n`, and sends those messages to a channel
+// newOutput returns a new decoder output
+func NewOutput(content []byte, contentLen int) *Output {
+	return &Output{
+		Content:    content,
+		ContentLen: contentLen,
+	}
+}
+
+// newOutputStop returns a new decoder output stop
+func newOutputStop() *Output {
```

Comment:
nit: I'd rename `newOutputStop` -> `newStopOutput`, to preserve English ordering